src/explorer/AzureAccountTreeItem.ts

```diff
@@ -2,17 +2,203 @@
+import * as vscode from "vscode";
+import { AzExtParentTreeItem, AzExtTreeItem, registerEvent, GenericTreeItem, ISubscriptionContext } from '@microsoft/vscode-azext-utils';
+import { AzureSubscriptionHelper } from "../azure/azureLogin/subscriptions";
+import { AzureSessionProvider, ReadyAzureSessionProvider, SelectionType, SignInStatus } from "../azure/azureLogin/authTypes";
+import { GeneralUtils } from "../utils/generalUtils";
+import { AzureAuth } from "../azure/azureLogin/azureAuth";
+import { APIMAccountCommandId } from "../constants";
+import { Subscription } from "@azure/arm-resources-subscriptions";
+import { createSubscriptionTreeItem } from "./ApiManagementProvider";
 
-import { ISubscriptionContext } from '@microsoft/vscode-azext-utils';
-import { AzureAccountTreeItemBase } from '@microsoft/vscode-azext-azureutils';
-import { ApiManagementProvider } from './ApiManagementProvider';
+export function createAzureAccountTreeItem(
+    sessionProvider: AzureSessionProvider,
+): AzExtParentTreeItem & { dispose(): unknown } {
+    return new AzureAccountTreeItem(sessionProvider);
+}
 
-export class AzureAccountTreeItem extends AzureAccountTreeItemBase {
-    public constructor(testAccount?: {}) {
-        super(undefined, testAccount);
+export class AzureAccountTreeItem extends AzExtParentTreeItem {
+    private subscriptionTreeItems: AzExtTreeItem[] | undefined;
+    public static contextValue: string = "azureApiManagementAzureAccount";
+    public readonly contextValue: string = AzureAccountTreeItem.contextValue;
+    constructor(private readonly sessionProvider: AzureSessionProvider) {
+      super(undefined);
+      this.autoSelectInTreeItemPicker = true;
+  
+      const onStatusChange = this.sessionProvider.signInStatusChangeEvent;
+      const onFilteredSubscriptionsChange = AzureSubscriptionHelper.getFilteredSubscriptionsChangeEvent();
+      registerEvent("azureAccountTreeItem.onSignInStatusChange", onStatusChange, (context) => this.refresh(context));
+      registerEvent("azureAccountTreeItem.onSubscriptionFilterChange", onFilteredSubscriptionsChange, (context) =>
+        this.refresh(context),
+      );
     }
-
-    public createSubscriptionTreeItem(root: ISubscriptionContext): ApiManagementProvider {
-        return new ApiManagementProvider(this, root);
+  
+    public override get label() {
+      return UiStrings.AzureAccount;
     }
-}
+  
+    public dispose(): void { }
+  
+    public hasMoreChildrenImpl(): boolean {
+      return false;
+    }
+  
+    // no need to sort the array
+    public compareChildrenImpl(item1: AzExtTreeItem, item2: AzExtTreeItem): number {
+      item1;
+      item2;
+      return 0;
+    }
+  
+    public async loadMoreChildrenImpl(): Promise<AzExtTreeItem[]> {
+      const existingSubscriptionTreeItems: AzExtTreeItem[] = this.subscriptionTreeItems || [];
+      this.subscriptionTreeItems = [];
+  
+      switch (this.sessionProvider.signInStatus) {
+        case SignInStatus.Initializing:
```

Comment:
default case?

Reply:
other case just return, but default case should do follow-up logic